Migrate service tidying & tests

PaulAdamDavis opened the pull request:
- Removes the unused migrate iframe darkMode property. This has not been implemented in the self-service migration tools, so it's just noise right now.
- Moves the migrate iframe payload generation to the migrate service. This moves payload generation to be in the same place the token is generated, so the gh-migrate-iframe.js only needs to handle the communication between the iframe and Ghost.
- Catch migrate iframe postMessage errors: If there's any error when posting a message to the migrate iframe, close the iframe and show a friendly user notification

PaulAdamDavis committed:
Remove unused migrate iframe darkMode property

This has not been implemented in the self-service migration tools, so it's just noise right now.

PaulAdamDavis committed:
Move migrate iframe payload generation to migrate service

This moves payload generation to be in the same place the token is generated, so the gh-migrate-iframe.js only needs to handle the communication between the iframe and Ghost.

PaulAdamDavis committed:
Merge branch 'main' into migrate-service-tidying

ErisDS reviewed (commented):
I think this is a great improvement. The Rabbit found one more change worth making :)

PaulAdamDavis committed:
Catch migrate iframe postMessage errors

If there's any error when posting a message to the migrate iframe, close the iframe and show a friendly user notification
